### PR TITLE
JavaScript Class vs Prototype

### LINKED ISSUE #2962: Prototype vs JavaScript Class
### Proposed title of article
Prototype vs JavaScript Class

### Introduction paragraph (2-3 paragraphs):
JavaScript classes, introduced in ECMAScript 2015, are primarily syntactical sugar over JavaScript's existing prototype-based inheritance.
The most important difference between class- and prototype-based inheritance is that a class defines a type which can be instantiated at runtime, whereas a prototype is itself an object instance.

A child of an ES6 class is another type definition which extends the parent with new properties and methods, which in turn can be instantiated at runtime. A child of a prototype is another object instance which delegates to the parent any properties that aren’t implemented on the child.

In this tutorial we dive into JavaScript to understand key differentiators between use of a Class and a Prototype - plus the impact each has on Lightning Web Components.

### Key takeaways: 
1.The Prototype Concept in JavaScript
2.The JavaScript Class
3.JavaScript Class v Prototype
4.Impact on Lightning Web Components

### PR DESCRIPTION
**Pull request by owinowendy**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
This fixes #2962

### CODE REVIEW

**Comment by jumamiller** [comment]
@owinowendy, add the hero image to this article!

**Comment by owinowendy** [comment]
Hi,@jumamiller , I've added the hero image! Thanks

**Comment by jumamiller** [comment]
@owinowendy ,thanks, I will proceed with review right away!

**Comment by hectorkambow** [comment]
plagiarism check done

File: content/articles/js-prototypes-vs-classes/index.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+- [Objectives](#objectives)
+- [Getting started with JavaSCript prototypes](#getting-started-with-javascript-prototypes)
+- [Getting started with JavaScript classes](#getting-started-with-javascript-classes)
+- [The key differences between JavaScript classes and prototypes](#the-key-differences-between-javascript-classes-and-prototypes)
+- [Conclusion](#conclusion)
+
+### Prerequisites
+
+To follow along with this tutorial, a basic understanding of JavaScript is required. Additionally, we will be using ECMAScript 2015.
+
```
Comment (jumamiller) [inline content/articles/js-prototypes-vs-classes/index.md:20]: Kindly have a proper article requirements

File: content/articles/js-prototypes-vs-classes/index.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    
+    </body>
+</html>
+```
+
+As you may have noticed, we created a `Student` object that we can then use to create other objects. It acts as our base object. To prove this, we have gone a step ahead and created the `stanfordStudent` instance as shown in the code above.  
+
+When we execute this code, the following output is generated:  
+
+![/engineering-education/js/prototype-output.png]
```
Comment (jumamiller) [inline content/articles/js-prototypes-vs-classes/index.md:73]: kindly fix this image

File: content/articles/js-prototypes-vs-classes/index.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+### Getting started with JavaScript classes
+
+JavaSCript classes, introduced on June 2, 2015, are no different from classes you might have already come across. They are used as templates for creating objects.
+
+> It's important to note that these classes are built on top of prototypes.
+
+ In this section, we convert the above example to the class and it would be very key for you to consider the key differences.
```
Comment (jumamiller) [inline content/articles/js-prototypes-vs-classes/index.md:81]: Clarify which example in question are you asking user to go through.

File: content/articles/js-prototypes-vs-classes/index.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+ One way to define classes in JavaSCript is via class declaration as shown below:  
+
+```javascript
+<html>
```
Comment (jumamiller) [inline content/articles/js-prototypes-vs-classes/index.md:86]: Kindly explain the code snippet in this article

File: content/articles/js-prototypes-vs-classes/index.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    
+    </body>
+</html>
+```
+
+### The key differences between JavaScript classes and prototypes
+
+In the previous sections, we've looked at the JavaScript classes and prototypes, in this part, let's have a look at their differences.
+
+The first and the main point is that JavaScript is based on **prototypes** and not classes. The prototypes themselves are objects instances as have in the previous examples while classes have types that are only instantiated at runtime. This explains why JavaScript developers should only stick to prototypes that using classes.
```
Comment (jumamiller) [inline content/articles/js-prototypes-vs-classes/index.md:128]: You have a breaking sentence on this paragraph. Kindly fix.

File: content/articles/js-prototypes-vs-classes/index.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+  }
+}
+
+const classyStudent = new ClassicalStudent("John Doe", "ExampleSchool")
+
+console.log(classyStudent.studentProto())
+
+```
+
+You quickly see a  prototypal inheritance in the first example, even though, it's more expressive in the second implementation. Nevertheless, you have seen how you can solve a problem using either case.
```
Comment (jumamiller) [inline content/articles/js-prototypes-vs-classes/index.md:170]: Kindly refrain from incomplete sentences. Be as expressive as possible. Thanks.

File: content/articles/js-prototypes-vs-classes/index.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+
+### Conclusion
+
+In this tutorial, we've seen how classes and prototypes are being used in JavaScript. We've also seen that prototypes themselves are object instances while classes only get instantiated at runtime.  
+With this background knowledge on how both features work in JavaScript, you can now decide on when to use either of the two.
```
Comment (jumamiller) [inline content/articles/js-prototypes-vs-classes/index.md:175]: I think you've not exhaustively discussed the key differences. Kindly add them. Thanks.

**Review by jumamiller — CHANGES_REQUESTED** [review]
Hi @owinowendy, I have suggusted a few changes. Kindly fix before we proceed to final review. Thanks

**Comment by owinowendy** [comment]
> Hi @owinowendy, I have suggusted a few changes. Kindly fix before we proceed to final review. Thanks

@jumamiller, changes done!

**Comment by jumamiller** [comment]
@owinowendy, I've made minor changes on your article. Ensure you go over the edits to better your future articles. Thanks.

**Comment by hectorkambow** [comment]
@owinowendy 

After some careful consideration, it struck us that this article may be lacking in substance and/or saturated throughout other blog sites and official documentations. 

We typically refrain from publishing content that is covered widely on the net or other blogs. We're more interested in original, practitioner-focused content that takes a deeper dive into programming-centric concepts. 

But in order to publish an article it has to serve sufficient value to the larger developer community at large. 
Please feel free to suggest an alternate topic to explore. 🚀

**Comment by owinowendy** [comment]
Hi @LinusMuema , why was the topic approved? It's really an interesting topic that has not been covered widely on the internet? It's quite unfair after the whole process only to be closed after final review. If there is a way I can rework on it, it would be better, BUT I"M SURE THIS TOPIC HAS NOT BEEN WIDELY COVERED ON THE INTERNET AS CLAIMED.

cc @hectorkambow

**Comment by LinusMuema** [comment]
Hi @owinowendy 
This article was closed because it did not have enough content. Here are some factors that led to that decision.

- The article is short. It hasn't reached the 750 words limit required while you still checked it above. I will attach the word count I did. Still on the same issue, there was a lot of repetition which is a sign that you were trying to add more words in the article but the limit wasn't reached.
Screenshot from 2021-08-19 09-23-38

- The value of the article. In my opinion as a javascript developer, this article doesn't really provide enough information or differences about the two that you are talking about. I have found some articles which have even gone to describe them in an OOP concept and involve other languages. So someone will definitely go for an article with more to offer. You keep on describing what each component is despite the fact that you already did that at the start of the article. Check [line 148](https://github.com/section-engineering-education/engineering-education/pull/3182/files#diff-7589fd475dc5cbcc1faa8b157a1c634999e2731b237658e130fbd170a61ace5dR148) and [line 185](https://github.com/section-engineering-education/engineering-education/pull/3182/files#diff-7589fd475dc5cbcc1faa8b157a1c634999e2731b237658e130fbd170a61ace5dR185) and see the similarity. I could still point out more instances of repetition. I even removed most of them after my final review

- The last issue, I did not point this out because the article was closed but, your hero image was a square. We require the image to be 600 x 400. Please check this for future articles. It doesn't end there. After a deeper research, I found that the image was also used in a youtube video explaining the same topic as this article which means you did not check the copyright issues on the image. This can lead to people suing section for using their material without consent. You can use unsplash or pexels to get these hero images or even create some yourself. I think @ahmadmardeni1 had a solution for this in the eng-ed channel. 
Screenshot from 2021-08-19 09-32-09

As you can see @owinowendy, your article has more than the issue described by @hectorkambow 
To answer your question, your article was approved because we expected more content than this in the article. Please take note of these issues in future articles to avoid this.

**Comment by owinowendy** [comment]
I think I can rework to add more value to it?

**Comment by owinowendy** [comment]
Codes are also words, what's the point of being wordy and not showing reader what to do?

**Comment by ahmadmardeni1** [comment]
About the hero image part, please use [this template](https://github.com/ahmadmardeni1/canva) in your future articles. @owinowendy

**Comment by hectorkambow** [comment]
> Hi @LinusMuema , why was the topic approved? It's really an interesting topic that has not been covered widely on the internet? It's quite unfair after the whole process only to be closed after final review. If there is a way I can rework on it, it would be better, BUT I"M SURE THIS TOPIC HAS NOT BEEN WIDELY COVERED ON THE INTERNET AS CLAIMED.
> 
> cc @hectorkambow

Thank you for the feedback - and we agree - we do our best to approve topics that have the greatest potential to get published. 

We try to flush out as much detail and substance from the beginning to avoid these kind of issues - at the same time - we have to be very cognisant of the of our entire reader community and ensure we are providing the best value possible for our larger developer community. 

Of course we encourage that the article be reworked - approached from a different perspective - or having it address a certain pain point that has not been explored before. 

If it related to this topic - we can reopen - or it becomes a new topic - we encourage a new topic form and PR.

**Pull request closed without merging**